iOS_SDK/OneSignalSDK/Source/OSMigrationController.m

```diff
@@ -0,0 +1,46 @@
+/**
+Modified MIT License
+
+Copyright 2020 OneSignal
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+1. The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+2. All copies of substantial portions of the Software may only be used in connection
+with services provided by OneSignal.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#import <Foundation/Foundation.h>
+#import "OneSignal.h"
+#import "OSMigrationController.h"
+#import "OSIndirectInfluence.h"
+
+@implementation OSMigrationController
+
+- (void)migrate {
+    [self migrateToVersion_02_14_00_AndGreater];
+}
+
+/**
+ * Support renaming of decodable classes for cached data
+ */
+- (void)migrateToVersion_02_14_00_AndGreater {
+    [NSKeyedUnarchiver setClass:[OSIndirectInfluence class] forClassName:@"OSIndirectNotification"];
```

Comment:
Does this actually migrate the data or just use the old name for reading and writing?

If this is just to keep the old name can this just live in the `OSIndirectInfluence` class itself?

Reply:
I see, since the key being saved and loaded isn't changing the data itself isn't being migrated. I think we should either migrate the data here or move this name mapping to `OneSignalUserDefaults.initStandard` since this is where the mapping is being applied.

Do we also need to call `NSKeyedArchiver` so the old name is also used when saving?